--- v8.h
+++ v8.h
@@ -2810,6 +2810,13 @@ class V8EXPORT V8 {
   static const int kMinimumSizeForLogLinesBuffer = 2048;
 
   /**
+   * Assign the current thread an id 
+   *
+   * This method can used to make sure a thread has an id, so it possible to call TerminateExecution for a perticular thread
+   */
+  static void AssignThreadId();
+  
+  /**
    * Retrieve the V8 thread id of the calling thread.
    *
    * The thread id for a thread should only be retrieved after the V8
